### DISCUSSION TITLE
Report Furnace/Chiller Stage?

### DISCUSSION THREAD

**Discussion by Peter847** [dahlb/ha_carrier#52] [Ideas]
Many thanks for this integration!!  

I have just finished writing a driver to interface a Hubitat hub to the Infinity system using Carrier's System Access Module - it provides an ASCII RS232 interface.  If I query the system mode the reply is AUTO/HEAT/COOL/OFF followed by the heating or cooling stage - 1/2/3.  Looking through the config download I do not see a stage setting passed back by Carrier, do they expose it on their API?

**Comment by dahlb** [comment]
"outdoor_unit_operational_status": self._updater.carrier_system.status.outdoor_unit_operational_status,
"indoor_unit_operational_status": self._updater.carrier_system.status.indoor_unit_operational_status,

outdoor status lists the cooling stages
indoor lists the heating stages

though I'd guess a heat pump would list both as outdoor status

https://github.com/dahlb/ha_carrier/blob/main/custom_components/ha_carrier/climate.py#L348C13-L349C114

**Reply by Peter847** [reply]
Great, many thanks.

I have to admit I am new to Home Assistant and very rusty on Python.  I'm guessing I need to expose those status values to HA by adding a couple of entities to the thermostat device?  Any help on the steps to do that would be a great help.

For background info, I need to add a duct booster fan to my HVAC system but only want it to switch on when either the furnace or chiller are actually operating rather than the fan just circulating.  Hence the need to know the stages.

**Comment by dahlb** [comment]
you might find _status_zone.conditioning easier as it is a single field that tells when it is heating/cooling vs idle

https://github.com/dahlb/ha_carrier/blob/main/custom_components/ha_carrier/climate.py#L167

**Comment by Peter847** [comment]
Yep, saw that and I will probably use it.
